### ISSUE TITLE
Add a new method to MathMLElement interface

### ISSUE THREAD

**Issue by NSoiffer** [mathml-refresh/mathml#231]
Having written a bunch of "polyfills" for MathML, there were three pieces of missing functionality that required hacks to implement: find the height/depth of an element, convert a CSS length value to pixels (e.g., '3em'), and clone an element including it's shadow root. The later two issues are not specific to MathML, but MathML is somewhat unique in defining and using an element's height above the baseline and depth below the baseline.

Rather than having to rely on a hack, MathML elements should provide a method to get this information. This could be an extension of `getBoundingClientRect` that adds baseline info or depth (since height above baseline is easily computed from that) or it could be an entirely new method that returns height/depth/width info (the later often being useful when doing computations involving box dimensions.

**Comment by NSoiffer** [comment]
At the Nov 1 meeting, we agreed this is potentially important to accessibility. @NSoiffer will write some text in a note in the core spec with @bkardell @fred-wang to review it. The note should mention the accessibility angle and solicit input from relevant parties.

**Comment by bkardell** [comment]
I'm not sure I understand why/what you're suggesting MathML should provide here and why it should be particular to MathML?  I'd like to not get back into co-evolution problems where we try to solve a thing in MathML that winds up being better solved elsewhere.  If this is about consulting box dimensions of some kind, it seems like it belongs to csswg?

**Comment by NSoiffer** [comment]
I agree that this is an issue that the CSS WG should address. However, since the need comes from manipulating MathML, it seems appropriate for us to come up with use cases and a proposal rather than just saying this is a problem and watching the issue get no attention. It is much easier to respond to a proposal than to a problem, even when the proposal is not a good one.

**Comment by ronkok** [comment]
I applaud this proposal, especially regarding the height/depth of an element relative to the baseline.

For a use case, I would like to implement the equivalent of the LaTeX `\vcenter` function. MathML Core does not currently enable this function. That's ok! It is used very infrequently. But if height and depth could be queried, then a `\vcenter` polyfill becomes feasible.
